driver/src/main/scala/org/mongodb/scala/ObservableImplicits.scala

```diff
@@ -370,6 +370,36 @@ trait ObservableImplicits {
       })
       promise.future
     }
+    
+     /**
+     * Returns the head option of the [[Observable]] in a [[scala.concurrent.Future]].
+     *
+     * @return the head option result of the [[Observable]].
+     */
+    def headOption(): Future[Option[T]] = {
+      val promise = Promise[Option[T]]()
+      observable.subscribe(new Observer[T]() {
+        @volatile
+        var subscription: Option[Subscription] = None
+
+        override def onError(throwable: Throwable): Unit = promise.tryFailure(throwable)
+
+        override def onSubscribe(sub: Subscription): Unit = {
+          subscription = Some(sub)
+          sub.request(1)
+        }
+
+        override def onComplete(): Unit = if(!promise.isCompleted) promise.trySuccess(None)
```

Comment:
There is no point checking for completeness of the promise. `IF THEN` is not atomic.